--- configure.in.old	2016-06-20 17:37:23.226006144 +0100
+++ configure.in	2016-06-20 17:37:55.658004769 +0100
@@ -364,6 +364,12 @@
 	aesni_obj=iaesx86.o
 	aesni_machine=x86
 	;;
+    x86_64-*-*-gnux32)
+    	# x86 assembly is slower but should run fine, since the 64 bit ABI
+	# assumes 64 bit integers, it may cause overflows.
+    	aesni_obj=iaesx86.o
+	aesni_machine=x32
+    	;;
     x86_64-*)
 	aesni_obj=iaesx64.o
 	aesni_machine=amd64
