--- configure.old	2016-06-20 16:25:46.320188215 +0100
+++ configure	2016-06-20 17:24:15.406039526 +0100
@@ -8793,6 +8793,12 @@
 	aesni_obj=iaesx86.o
 	aesni_machine=x86
 	;;
+    x86_64-*-*-gnux32)
+    	# x86 assembly is slower but should run fine, since the 64 bit ABI
+	# assumes 64 bit integers, it may cause overflows.
+    	aesni_obj=iaesx86.o
+	aesni_machine=x32
+    	;;
     x86_64-*)
 	aesni_obj=iaesx64.o
 	aesni_machine=amd64
